x-pack/platform/plugins/shared/lens/public/react_embeddable/initializers/initialize_alert_rules.tsx

```diff
@@ -0,0 +1,145 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { escapeRegExp } from 'lodash';
+import { ActionTypeRegistryContract } from '@kbn/alerts-ui-shared';
+import { ESQLControlVariable, apiPublishesESQLVariables } from '@kbn/esql-types';
+import { parse, Walker } from '@kbn/esql-ast';
+import { AggregateQuery } from '@kbn/es-query';
+import { i18n } from '@kbn/i18n';
+import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
+import { tracksOverlays } from '@kbn/presentation-containers';
+import { toMountPoint } from '@kbn/react-kibana-mount';
+import type { RuleTypeRegistryContract } from '@kbn/response-ops-rule-form';
+import { RuleFormFlyout } from '@kbn/response-ops-rule-form/flyout';
+import { isValidRuleFormPlugins } from '@kbn/response-ops-rule-form/lib';
+import { ES_QUERY_ID } from '@kbn/rule-data-utils';
+import React from 'react';
+import type {
+  LensAlertRulesApi,
+  LensCreateAlertRuleInitialValues,
+  LensEmbeddableStartServices,
+} from '../types';
+
+export function initializeAlertRules(
+  uuid: string,
+  parentApi: unknown,
+  startDependencies: LensEmbeddableStartServices
+): { api: LensAlertRulesApi } {
+  return {
+    api: {
+      createAlertRule: (
+        passedInitialValues: LensCreateAlertRuleInitialValues,
+        ruleTypeRegistry: RuleTypeRegistryContract,
+        actionTypeRegistry: ActionTypeRegistryContract
+      ) => {
+        const { coreStart } = startDependencies;
+        const ruleFormPlugins = {
+          ...startDependencies,
+          ...coreStart,
+          http: startDependencies.coreHttp,
+        };
+        const canShowRuleForm =
+          isValidRuleFormPlugins(ruleFormPlugins) && ruleTypeRegistry && actionTypeRegistry;
+
+        if (!canShowRuleForm) {
+          throw new Error(
+            i18n.translate('xpack.lens.createAlertRuleError', {
+              defaultMessage: 'Unable to open the alert rule form, some services are missing',
+            })
+          );
+        }
+
+        const esqlVariables$ = apiPublishesESQLVariables(parentApi)
+          ? parentApi.esqlVariables$
+          : undefined;
+        let initialValues = { ...passedInitialValues };
+
+        const esqlVariables = esqlVariables$?.getValue() ?? [];
+        if (esqlVariables.length) {
+          initialValues = {
+            params: {
+              ...initialValues.params,
+              esqlQuery: parseEsqlVariables(initialValues.params?.esqlQuery, esqlVariables),
+            },
+          };
+        }
+
+        const overlayTracker = tracksOverlays(parentApi) ? parentApi : undefined;
+
+        const closeRuleForm = () => {
+          overlayTracker?.clearOverlays();
+          handle?.close();
+        };
+
+        const handle = coreStart.overlays.openFlyout(
+          toMountPoint(
+            <KibanaContextProvider services={ruleFormPlugins}>
+              <RuleFormFlyout
```

Comment:
I'm having the following error multiple times when the panel opens in the console:

```
dataViewFields.map is not a function
```

Reply:
Tracked in https://github.com/elastic/kibana/issues/220235